src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ActionPlanService.kt

```diff
@@ -53,9 +55,36 @@ class ActionPlanService(
     return actionPlanRepository.save(draftActionPlan)
   }
 
+  fun submitDraftActionPlan(id: UUID, submittedByUser: AuthUser): ActionPlan {
+
+    val draftActionPlan = getDraftActionPlanAndThrowExceptionIfNotFound(id)
+    submitActionPlan(draftActionPlan, submittedByUser)
+    val savedDraftActionPlan = actionPlanRepository.save(draftActionPlan)
+    actionPlanEventPublisher.actionPlanSubmitEvent(savedDraftActionPlan)
+
+    return savedDraftActionPlan
+  }
+
+  fun getActionPlan(id: UUID): ActionPlan {
+    return actionPlanRepository.findById(id).orElseThrow {
+      throw EntityNotFoundException("action plan not found [id=$id]")
+    }
+  }
+
   private fun updateDraftActivityPlan(draftActionPlan: ActionPlan, update: ActionPlan) {
     update.numberOfSessions?.let {
       draftActionPlan.numberOfSessions = it
     }
   }
+
+  private fun getDraftActionPlanAndThrowExceptionIfNotFound(id: UUID): ActionPlan {
+    val draftActionPlan = actionPlanRepository.findByIdAndSubmittedAtIsNull(id)
+      ?: throw EntityNotFoundException("draft action plan not found [id=$id]")
+    return draftActionPlan
+  }
+
+  private fun submitActionPlan(draftActionPlan: ActionPlan, submittedByUser: AuthUser) {
```

Comment:
When I read the function name, I expected the action plan to be submitted here or something along those lines. I'm assuming this was done in the same way I named the update function so that may not be clear either. Just a note.

Reply:
I will change name to submitDraftActionPlan